src/components/Auth/CheckboxCustom.tsx

```diff
@@ -0,0 +1,50 @@
+import { Box, Checkbox, Flex, FormControl, FormErrorMessage, FormLabel, Text } from '@chakra-ui/react'
+import { ReactNode } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import useDarkMode from '~src/themes/saas/hooks/useDarkMode'
+
+export interface CustomCheckboxProps {
+  formValue: string
+  label: ReactNode
+  required?: boolean
+  colorScheme?: string
+}
+
+const CustomCheckbox = ({
+  formValue,
+  label,
+  required = false,
+
+  colorScheme = 'brandScheme',
+}: CustomCheckboxProps) => {
+  const { t } = useTranslation()
+  const { textColor } = useDarkMode()
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext()
+
+  const validationRules = required ? { required: { value: true, message: t('form.error.field_is_required') } } : {}
```

Comment:
Avoid to use ternary operators unless is absolutely needed.  Is much more readable if you do:

```
let validationRules = {}
if(required) {
    validationRules = { required: { value: true, message: t('form.error.field_is_required') } } 
}
```

Is much readable, try to use this method instead of the difficult to understand ternary opertaions